src/Result.php

```diff
@@ -112,4 +143,112 @@ public function getMode()
     {
         return $this->parser->mode;
     }
+
+    /**
+     * Cleanup the relations on a models array
+     *
+     * @param $models
+     * @return array
+     */
+    public function cleanupRelationsOnModels($models)
+    {
+        $response = [];
+
+        if ($models instanceof Collection) {
+            foreach ($models as $model) {
+                $response[] = $this->cleanupRelations($model);
+            }
+        }
+
+        return $response;
+    }
+
+    /**
+     * Cleanup the relations on a single model
+     *
+     * @param $model
+     * @return mixed
+     */
+    public function cleanupRelations($model)
+    {
+        if (!($model instanceof Model)) {
+            return $model;
+        }
+
+        // get the relations which already exists on the model (e.g. with $builder->with())
+        $allowedRelations = $this->getRelationsRecursively($model);
+
+        // parse the model to an array and get the relations which got added unintentionally
+        // (e.g. when accessing a relation in an accessor method or somewhere else)
+        $response = $model->toArray();
```

Comment:
I see your concerns.. The problem is that if we don't call ->toArray(), Laravel will and the relations get loaded again. I could move the cleanup from the getResult() method to getResponse() where you don't expect the raw models but a json response (which you'll still get). If you use getResult() directly, you would have to cleanup the data by yourself (or just call the public cleanup methods).